### PR TITLE
Restore previous browsing agent behavior when evaluating on WebArena and miniwob++ only

### PR DESCRIPTION
**Pull request by frankxu2004**
More context here: https://github.com/OpenDevin/OpenDevin/pull/2170#discussion_r1632174553

The original noop action to get initial observation is removed from a previous PR. This PR brings back the functionality to enable good support for web browsing benchmarks such as WebArena and MiniWOB.

For normal web browsing, we don't need to issue the noop since the browser always starts with a blank page, so the initial observation is non-existent anyway, and the browsing agent will first issue a goto(url) action to kick start the process.

However, for evaluation on benchmarks such as webarena and miniwob++, the browser env is initialized with some specific URL and the initial observation is quite crucial for the browsing agent to make the first decision - that's way I added the noop action first to get the initial observation.

### CODE REVIEW

**Review by enyst — APPROVED** [review]
Thank you for the fix! And the explanations. I completely missed this point.

With this explicit EVAL_MODE, this shouldn't interfere with agent as delegate. No matter what tweaks we'll want in the future, it's clear what the behavior needs to be. Thanks for your patience :)

File: agenthub/browsing_agent/browsing_agent.py [hunk 1]
```diff
@@ -118,6 +123,14 @@ def step(self, state: State) -> Action:
         last_obs = None
         last_action = None
 
+        print(state.history)
```
Comment (enyst) [inline agenthub/browsing_agent/browsing_agent.py:126]:
```suggestion
```

**Comment by enyst** [comment]
@frankxu2004 what do you think, maybe eval_mode is a potential candidate for https://github.com/OpenDevin/OpenDevin/issues/2337 ?

**Comment by frankxu2004** [comment]
> @frankxu2004 what do you think, maybe eval_mode is a potential candidate for #2337 ?

Yes it definitely is! Right now I am using environment variables for config, it works as most of the time we don't need to set anything other than default, but it'd be nice to have something more elegant

**Pull request merged**